src/java/org/apache/cassandra/config/Config.java

```diff
@@ -1149,6 +1151,13 @@ public enum DiskAccessMode
         standard,
     }
 
+    public enum CommitLogDiskAccessMode
+    {
+        standard,
```

Comment:
I see three problems with the `standard` option:
- there is no implementation of standard mode for uncompressed log (we probably can use [this simple one](https://github.com/datastax/cassandra/blob/ds-trunk/src/java/org/apache/cassandra/db/commitlog/UncompressedSegment.java) if we need to have it);
- the current code requires the user to change from `direct` to `standard` when enabling compression/encryption, which is not nice, and would be a deal breaker for `direct` being the default;
- we understand `mmap` to silently translate to `standard` for compressed/encrypted logs, which is legacy behaviour that we do not want to change, but which is at odds with the above.

The simplest way to solve this is to explain in the option's comment that it currently only applies to uncompressed/unencrypted logs and remove `standard` from this enum. But if we are planning also to implement direct writing of compressed/encrypted logs (valuable in its own right), it look like we should change the alternative to `mmap_or_standard` or maybe something like `legacy`.

Reply:
I'm still at the position we should have one enum which cover all the needed cases and just validate that for certain uses.

Here in particular, it would be useful to add, as suggested by @blambov - `legacy` or assume undefined value would denote that. On the other hand, it would be useful to add `auto` for the new, optimized behavior. 

When access mode is specified explicitly, and it is invalid for certain use, for example - `direct` for compressed/encrypted commitlog, I would like to fail the startup with configuration error rather than do something implicitly. In other words, I wish Cassandra does not configure something implicitly unless we explicitly specify something that suggest default, automatic or legacy behavior.

That applies to both commitlog and sstable data files, and maybe would be useful to separate that configuration for read and write.